pkg/util/template/resource.go

```diff
@@ -144,16 +137,66 @@ func (t languageTemplate) GetLanguageTag() string {
 
 var templateLanguageTagRegex = regexp.MustCompile("^templates/([a-zA-Z0-9-_]+)/")
 
-func mergeTemplates(layers []resource.LayerFile, args map[string]interface{}) (*resource.MergedFile, error) {
-	preferredLanguageTags, _ := args[ResourceArgPreferredLanguageTag].([]string)
-	defaultLanguageTag, _ := args[ResourceArgDefaultLanguageTag].(string)
+func viewTemplates(resources []resource.ResourceFile, rawView resource.View) (interface{}, error) {
+	switch view := rawView.(type) {
+	case resource.EffectiveResourceView:
+		return viewTemplatesEffectiveResource(resources, view)
+	case resource.EffectiveFileView:
+		return viewTemplatesEffectiveFile(resources, view)
+	default:
+		return nil, fmt.Errorf("unsupported view: %T", rawView)
+	}
+}
+
+func viewTemplatesEffectiveFile(resources []resource.ResourceFile, view resource.EffectiveFileView) (interface{}, error) {
+	// When template is viewed as EffectiveFile,
```

Comment:
I guess this is related to how we handle multilingual templates on portal. If default language tag is `zh`, then we should not allow the user to provide templates for `zh`.

Reply:
However developers can list all resources in FS through portal API, so we may have some localized resources that can be accessed, yet cannot be updated/deleted.

I suppose this would not have much impact, since it should not affect the functionality of portal frontend. Let's ignore that and just forbid updating localized resources for default language?

Reply:
I guess the root problem is having `__default__`. Let's see if we can remove that in https://github.com/authgear/authgear-server/issues/717